src/lib.rs

```diff
@@ -44,6 +45,41 @@ unsafe fn strlen(p: *const c_char) -> usize {
     n
 }
 
+/// A drop-in implementation of `memchr` that is const.
+///
+/// This is expected to perform way worse than any version the `memchr` crate provides.
+#[cfg(feature = "nightly")]
+const fn memchr_const(needle: u8, haystack: &[u8]) -> Option<usize> {
+    let mut i = 0;
+    loop {
+        i += 1;
+        if haystack.len() == i {
+            return None;
+        }
+        if haystack[i] == needle {
+            return Some(i);
+        }
+    }
+}
+
+/// Wrapper around memchr that is const, but still fast at runtime by dispatching through
+/// const_eval_select.
+#[inline]
+#[cfg(feature = "nightly")]
+const fn memchr(needle: u8, haystack: &[u8]) -> Option<usize> {
+    // unsafe: Both versions provide the same functionality
+    unsafe {
+        core::intrinsics::const_eval_select(
+            (needle, haystack),
+            memchr_const,
+            memchr::memchr,
+            )
```

Comment:
This seems to be missing a rustfmt pass.

Reply:
Changed in the updated version. (Selectively: A full `cargo fmt` fun would produce changes to unrelated portions of the code.)